File: DGV7.0-NON-SAAS/PayHub-Guest-iOS/PayHubGuest/Models/ApiModels.swift
import Foundation

struct NetworkDetectEntry: Codable {
    let phone: String
    let network: String
}

struct NetworkDetectResponse: Codable {
    let status: String?
    let network: String?
    let data: [NetworkDetectEntry]?
}

struct VerifyCustomerResponse: Codable {
    let status: String
    let desc: String?
    let customerName: String?
    let customerAddress: String?

    enum CodingKeys: String, CodingKey {
        case status, desc
        case customerName = "customer_name"
        case customerAddress = "customer_address"
    }
}

struct CheckoutInitResponse: Codable {
    let status: String
    let reference: String?
    let amount: Double?
    let checkoutUrl: String?
    let desc: String?

    enum CodingKeys: String, CodingKey {
        case status, reference, amount, desc
        case checkoutUrl = "checkout_url"
    }
}

struct GuestOrderStatusResponse: Codable {
    let ref: String?
    let status: String?
    let service: String?
    let amount: String?
    let desc: String?
    let responseDesc: String?
    let meterNumber: String?
    let token: String?
    let tokenUnit: String?
    let customerId: String?

    enum CodingKeys: String, CodingKey {
        case ref, status, service, amount, desc, token
        case responseDesc = "response_desc"
        case meterNumber = "meter_number"
        case tokenUnit = "token_unit"
        case customerId = "customer_id"
    }
}

/// Local-only receipt snapshot used for the Receipt screen and PDF/image export.
/// Guest Mode stores no server-side transaction history — this is the only durable
/// copy of a purchase, generated client-side right after payment.
struct GuestReceipt {
    let reference: String
    let service: String
    let recipient: String
    let amountPaid: Double
    let status: String
    let date: Date
    var meterNumber: String? = nil
    var token: String? = nil
    var tokenUnit: String? = nil
}

/// A generic {"status":"failed","desc":"..."} error body shape used across web/guest-api/*.php.
struct ApiFailure: Codable {
    let status: String?
    let desc: String?
}

enum ApiResult<T> {
    case success(T)
    case failure(String)
}

File: DGV7.0-NON-SAAS/PayHub-Guest-Android/app/src/main/java/com/payhub/guest/data/model/ApiModels.kt
package com.payhub.guest.data.model

import com.google.gson.annotations.SerializedName

data class ApiFailure(
    val status: String? = null,
    val desc: String? = null,
)

data class NetworkDetectResponse(
    val status: String? = null,
    val network: String? = null,
    val data: List<NetworkDetectEntry>? = null,
)

data class NetworkDetectEntry(
    val phone: String,
    val network: String,
)

data class VerifyCustomerResponse(
    val status: String,
    val desc: String? = null,
    @SerializedName("customer_name") val customerName: String? = null,
    @SerializedName("customer_address") val customerAddress: String? = null,
)

data class CheckoutInitResponse(
    val status: String,
    val reference: String? = null,
    val amount: Double? = null,
    @SerializedName("checkout_url") val checkoutUrl: String? = null,
    val desc: String? = null,
)

data class GuestOrderStatusResponse(
    val ref: String? = null,
    val status: String? = null,
    val service: String? = null,
    val amount: String? = null,
    val desc: String? = null,
    @SerializedName("response_desc") val responseDesc: String? = null,
    @SerializedName("meter_number") val meterNumber: String? = null,
    val token: String? = null,
    @SerializedName("token_unit") val tokenUnit: String? = null,
    @SerializedName("customer_id") val customerId: String? = null,
)

/** Local-only receipt snapshot used for the Receipt screen and locally-cached History list. */
data class GuestReceipt(
    val reference: String,
    val service: String,
    val recipient: String,
    val amountPaid: Double,
    val status: String,
    val dateMillis: Long,
    val meterNumber: String? = null,
    val token: String? = null,
    val tokenUnit: String? = null,
)
